Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebPage.cpp

```diff
@@ -179,30 +189,46 @@ class PageLoaderClient final : public API::InjectedBundle::PageLoaderClient {
 
     void didStartProvisionalLoadForFrame(WebPage&, WebFrame& frame, RefPtr<API::Object>&) override
     {
-        if (!frame.isMainFrame())
-            return;
-        webkitWebPageSetURI(m_webPage, getDocumentLoaderURL(frame.coreFrame()->loader().provisionalDocumentLoader()));
+        const auto uri = getDocumentLoaderURL(frame.coreFrame()->loader().provisionalDocumentLoader());
+
+        if (auto* webKitFrame = webkitFrameGet(&frame))
+            webkitFrameSetURI(webKitFrame, uri);
+
+        if (frame.isMainFrame())
+            webkitWebPageSetURI(m_webPage, uri);
```

Comment:
I see now that we can avoid the uri creation if it's not main frame and it doesn't exist. So, we could do something like:

```
auto* webKitFrame = webkitFrameGet(&frame);
if (!webkitFrame && !frame.isMainFrame())
    return;

const auto uri = getDocumentLoaderURL(frame.coreFrame()->loader().provisionalDocumentLoader());
...
```